modules/core/src/main/scala/gem/parser/Angle.scala

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2016-2017 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package gem
+package parser
+
+import atto._, Atto._
+import gem.math.{ Angle, HourAngle }
+
+/** Parsers for `[[gem.math.Angle]].` */
+trait AngleParsers {
+  import MiscParsers._
+
+  /**
+   * Generic parser for the components of an angle in "11 22 33.444555" format, with at least 1
+   * and at most 6 digits following the decimal point, and terminal parsers for each segment.
+   */
+  def genAngle(t1: Parser[_], t2: Parser[_], t3: Parser[_]): Parser[(Int, Int, Int, Int, Int)] =
+    for {
+      h  <- int <~ t1
+      m  <- int <~ t2
+      s  <- int <~ char('.')
+      µs <- frac(6)
+      _  <- t3
+    } yield (h, m, s, µs / 1000, µs % 1000)
```

Comment:
The old `(a |@| b |@| c)(f)` syntax was bothersome because operator precedence of `|@|` wasn't low enough. But the new `(a, b, c).mapN(f)` syntax doesn't have that issue and it works well. So I changed a bunch of `for` comps.